niceman/distributions/redhat.py

```diff
@@ -0,0 +1,341 @@
+            package_specs.append(package_spec)
+
+        # Doing in one shot to fail early if any of the versioned specs
+        # couldn't be satisfied
+        lgr.debug("Installing %s", ', '.join(package_specs))
+        session.execute_command(
+            # TODO: Pull env out of provenance for this command.
+            ['yum', 'install', '-y'] + package_specs,
+            # env={'DEBIAN_FRONTEND': 'noninteractive'}
+        )
+
+    def satisfies_package(self, package):
+        """return True if this distribution (self) satisfies the requirements 
+        of the passed package"""
+        if not isinstance(package, Package):
+            raise TypeError('satisfies_package() requires a package argument')
+        if not isinstance(package, RPMPackage):
+            return False
+        return any([p.satisfies(package) for p in self.packages])
+
+    def satisfies(self, other):
+        """return True if this distribution (self) satisfies the requirements 
+        of the other distribution (other)"""
+        if not isinstance(other, Distribution):
+            raise TypeError('satisfies() requires a distribution argument')
+        if not isinstance(other, RedhatDistribution):
+            return False
+        return all(map(self.satisfies_package, other.packages))
+
+    def __sub__(self, other):
+        # the semantics of distribution subtraction are, for d1 - d2:
+        #     what is specified in d1 that is not specified in d2
+        #     or how does d2 fall short of d1
+        #     or what is in d1 that isn't satisfied by d2
+        return [p for p in self.packages if not other.satisfies_package(p)]
+
+_register_with_representer(RedhatDistribution)
+
+
+class RPMTracer(DistributionTracer):
+    """.rpm-based (and using yum and rpm) systems package tracer
+    """
+
+    # The Redhat tracer is not designed to handle directories
+    HANDLES_DIRS = False
+
+    def _init(self):
+        # TODO: we might want a generic helper for collections of things
+        # where we could match based on the set of attrs which matter
+        self._package_install_dates = {}
+
+    def identify_distributions(self, files):
+        """
+        Return a distribution object containing package and repos source
+        information for files found to be from CentOS distribution
+
+        Parameters
+        ----------
+        files : list(string)
+            List of files to check to see if they come from CentOS packages
+
+        Returns
+        -------
+        dist : RedhatDistribution object
+        remaining_files : list(string)
+            List of files not identified to be in CentOS packages
+        """
+        if not files:
+            return
+
+        try:
+            redhat_version = self._session.read('/etc/redhat-release').strip()
+            self._session.exists('/etc/os-release')
+            _, _ = self._session.execute_command(
+                ['grep', '-i', '^ID.*="rhel fedora"', '/etc/os-release'])
```

Comment:
seems to be not a good/reliable indicator on old(er) systems?
```
(niceman-dev) [yhalchen@discovery niceman]$ lsb_release  -r
Release:        6.9
(niceman-dev) [yhalchen@discovery niceman]$ rpm
RPM version 4.8.0
...
(niceman-dev) [yhalchen@discovery niceman]$ rpm -qf /bin/bash
bash-4.1.2-48.el6.x86_64
(niceman-dev) [yhalchen@discovery niceman]$ grep -i '^ID.*="rhel fedora"' /etc/os*
grep: /etc/os*: No such file or directory
```
may be 
```shell
(niceman-dev) [yhalchen@discovery niceman]$ cat /etc/redhat-release 
CentOS release 6.9 (Final)
```
would provide a more stable indicator?  since you are just ignoring the output anyways -- why bother after you already checked for /etc/redhat-release ?